source/grid/tria.cc

```diff
@@ -6732,34 +6737,55 @@ namespace internal
                       }
                   }
 
+                  unsigned int chosen_line_tetrahedron = 0;
                   // set up new lines
-                  {
-                    static constexpr dealii::ndarray<unsigned int, 6, 2>
-                      new_line_vertices_hex = {{{{22, 26}},
-                                                {{26, 23}},
-                                                {{20, 26}},
-                                                {{26, 21}},
-                                                {{24, 26}},
-                                                {{26, 25}}}};
-
-                    static constexpr dealii::ndarray<unsigned int, 6, 2>
-                      new_line_vertices_tet = {{{{6, 8}},
-                                                {{X, X}},
-                                                {{X, X}},
-                                                {{X, X}},
-                                                {{X, X}},
-                                                {{X, X}}}};
-
-                    const auto &new_line_vertices =
-                      (reference_cell_type == ReferenceCells::Hexahedron) ?
-                        new_line_vertices_hex :
-                        new_line_vertices_tet;
+                  if (reference_cell_type == ReferenceCells::Hexahedron)
+                    {
+                      static constexpr dealii::ndarray<unsigned int, 6, 2>
+                        new_line_vertices = {{{{22, 26}},
+                                              {{26, 23}},
+                                              {{20, 26}},
+                                              {{26, 21}},
+                                              {{24, 26}},
+                                              {{26, 25}}}};
+                      for (unsigned int i = 0; i < n_new_lines; ++i)
+                        new_lines[i]->set_bounding_object_indices(
+                          {vertex_indices[new_line_vertices[i][0]],
+                           vertex_indices[new_line_vertices[i][1]]});
+                    }
+                  else if (reference_cell_type == ReferenceCells::Tetrahedron)
+                    {
+                      // in the tetrahedron case, we have the three
+                      // possibilities (6,8), (5,7), (4,9) -> pick the
+                      // shortest line to guarantee the best possible aspect
+                      // ratios
+                      static constexpr dealii::ndarray<unsigned int, 3, 2>
+                        new_line_vertices = {{{{6, 8}}, {{5, 7}}, {{4, 9}}}};
+
+                      const auto &vertices = triangulation.get_vertices();
+                      double      best_distance =
```

Comment:
I think the namings `best` and `tentative` could be improved here. E.g. replacing `best` by `min` this would automatically reflect the underlying definition that "best = min". (`tentative_distance` could be replaced by `current_distance` or `distance_i`).

Reply:
I agree, updated the naming of the variables to improve clarity.